src/ui/public/agg_types/metrics/lib/parent_pipeline_agg_helper.js

```diff
@@ -0,0 +1,61 @@
+import metricAggTemplate from 'ui/agg_types/controls/sub_agg.html';
+import _ from 'lodash';
+import VisAggConfigProvider from 'ui/vis/agg_config';
+import VisSchemasProvider from 'ui/vis/schemas';
+import { parentPipelineAggController } from './lib/parent_pipeline_agg_controller';
+import { parentPipelineAggWritter } from './lib/parent_pipeline_agg_writter';
+
+const ParentPipelineAggHelperProvider = function (Private) {
+
+  const AggConfig = Private(VisAggConfigProvider);
+  const Schemas = Private(VisSchemasProvider);
+
+  const metricAggFilter = ['!top_hits', '!percentiles', '!percentile_ranks', '!median', '!std_dev'];
+  const metricAggSchema = (new Schemas([
+    {
+      group: 'none',
+      name: 'metricAgg',
+      title: 'Metric Agg',
```

Comment:
Is `Metric Agg` specific to derivatives or a certain type of visualization?  I think it's only used for derivates right now, but the name is so generic, it sounds like it can be used elsewhere?  I guess I don't completely understand what a `metric agg` is.  In the UI, is it the Y-Axis `Aggregation` spot?  But only when it's not one of the aggregations listed above.  Like a `Count` is a `Metric Agg` but `Top Hits` is some other kind of `Agg`?  Or is only `derivative` a `metric agg`?